src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -475,7 +475,8 @@ void OTARequestor::TriggerImmediateQueryInternal()
 OTARequestorInterface::OTATriggerResult OTARequestor::TriggerImmediateQuery()
 {
     ProviderLocationType providerLocation;
-    if (mOtaRequestorDriver->GetNextProviderLocation(providerLocation) != true)
+    bool listExhausted = false;
+    if (mOtaRequestorDriver->GetNextProviderLocation(providerLocation, listExhausted) != true)
```

Comment:
nit: I wonder if the second param can be an optional param and not need to be passed in as most calls to `GetNextProviderLocation` doesn't look at this value? OR perhaps the return value can be an enum of {kFound, kNotFound, kFoundButExhausted}

Reply:
True- I did think about that. I cannot make it an optional parameter since it is being passed by reference. I could change the return value or overload the function. But at the end I decided mine was the least evil :sweat_smile: .